### PR TITLE
Store UUID of original element in @corresp

### PR DESCRIPTION
**Pull request by Monceber**
This change adds code that stores UUID of original element when it's being clones to the `@corresp` attribute (when applicable). This allow us to track back some of the elements, that are being cloned in the process and new IDs are generated for them as the result. In most cases these elements are KeySig/Clef/MeterSig/etc. (elements of the ScoreDef).

`GetElementAttr` is also changed to get original element whenever found object has @corresp set.

### CODE REVIEW

**Comment by ahankinson** [comment]
Is this still using UUIDs? (i.e., [UUIDv4](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random))) Or is it just generic XML IDs? IIRC the XML IDs assigned by Verovio are not actually UUIDs anymore...

**Review by lpugin — CHANGES_REQUESTED** [review]

**Comment by rettinghaus** [comment]
But it doesn't change the original MEI, does it? From my point of view this is a very MEI-ish way to not store additional information in the SVG output, but still have the ability to track down where clefs, grouping symbols etc. originate from in the toolkit. 
At the moment I don't see a better way for handling spanning control elements as it is implemented right now, but it is not very elegant.

File: src/object.cpp [hunk 1]
```diff
@@ -710,21 +710,10 @@ Object *Object::FindElementInLayerStaffDefsByUUID(const std::string &uuid)
...
-        if (layer->GetStaffDefClef() && (layer->GetStaffDefClef()->GetUuid() == uuid)) return layer->GetStaffDefClef();
-        if (layer->GetStaffDefKeySig() && (layer->GetStaffDefKeySig()->GetUuid() == uuid))
-            return layer->GetStaffDefKeySig();
-        if (layer->GetStaffDefMensur() && (layer->GetStaffDefMensur()->GetUuid() == uuid))
-            return layer->GetStaffDefMensur();
-        if (layer->GetStaffDefMeterSig() && (layer->GetStaffDefMeterSig()->GetUuid() == uuid))
-            return layer->GetStaffDefMeterSig();
-        if (layer->GetStaffDefMeterSigGrp() && (layer->GetStaffDefMeterSigGrp()->GetUuid() == uuid))
-            return layer->GetStaffDefMeterSigGrp();
+    for (Object *layer : layers) {
```
[the remark this answers is not in the document]
Comment (Monceber) [reply src/object.cpp:713]: Yes, I guess it does make sense to use existing Process() functionality. I'm not quite sure about storing `m_child` though - it is quite convenient to save it, but it seems to break the principle behind comparison functors that we have right now. I.e. none of them store any data at the moment, and we would be changing that here.

[the remark this answers is not in the document]
Comment (Monceber) [reply src/object.cpp:713]: Alternatively, we could go with Functor approach:

```c++
class FindLayerUuidWithinStaffDefParams : public FunctorParams {
public:
    explicit FindLayerUuidWithinStaffDefParams(const std::string &Uuid)
    {
        m_uuid = Uuid;
        m_object = NULL;
    }

    Object *m_object;
    std::string m_uuid;
};
```

Change `FindElementInLayerStaffDefsByUUID` to take these params, and in toolkit we'd have to call `Process()`:

```c++
Functor findByUuid(&Object::FindElementInLayerStaffDefsByUUID);
FindLayerUuidWithinStaffDefParams params(xmlId);
m_doc.GetDrawingPage()->Process(&findByUuid, &params);
```

This way we won't need to call `FindElementInLayerStaffDefsByUUID` twice and we search recursively through the tree.

[the remark this answers is not in the document]
Comment (lpugin) [reply src/object.cpp]: Even better! I like it.

**Review by lpugin — APPROVED** [review]

**Pull request merged**